dali/operators/reader/loader/video/frames_decoder.h

```diff
@@ -58,8 +58,13 @@ struct AvState {
       av_frame_free(&frame_);
     }
     avcodec_free_context(&codec_ctx_);
-    avformat_close_input(&ctx_);
-    avformat_free_context(ctx_);
+    if (ctx_ != nullptr) {
+      if (ctx_->pb != nullptr) {
+        avio_context_free(&ctx_->pb);
+      }
+      avformat_close_input(&ctx_);
```

Comment:
Based on https://ffmpeg.org/doxygen/trunk/demux_8c_source.html#l00369, 'avformat_close_input' seems to call `avformat_free_context` and `avio_close` which calls `avio_context_free`. Still please double check this in the version of FFmpeg we use (asan doesn't complain so I think we are not leaking anything).

Reply:
I did that based on the documentation that said that allocation of this context should be freed. Since in this scenario that we use it the wrapping object destructor takes care of it, I removed the call in our code.